Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/CytoscapeJsonCreator.java

```diff
@@ -89,4 +124,108 @@ public String getJson(Map<String, Node> nodes, List<Link> links) {
+            dataObj.setSourceReference(link.getSourceReference());
+            dataObj.setStatementIds(new ArrayList<String>());
+            dataObj.getStatementIds().add(link.getStatementId());
+            cytoscapeLinkList.add(createCytoscapeLinkObject(dataObj));
+            i++;
+        }
+        return cytoscapeLinkList;
+    }
+    
+    private CytoscapeNodeObject createCytoscapeNodeObject(CytoscapeNodeDataObject dataObj){
+        CytoscapeNodeObject obj = new CytoscapeNodeObject();
+        obj.setData(dataObj);
+        return obj;
+    }
+    
+    private CytoscapeLinkObject createCytoscapeLinkObject(CytoscapeLinkDataObject dataObj){
+        CytoscapeLinkObject obj = new CytoscapeLinkObject();
+        obj.setData(dataObj);
+        return obj;
+    }
+    
+    @Override
+    public String submitTransformationRequest(String conceptId, IProject project){
+        String conceptIdToken = conceptId.substring(conceptId.lastIndexOf('/') + 1);
+        if(!cache.isKeyInCache(conceptIdToken)){
+              Callable<PublicSearchObject> callable = () -> {
+
+                  PublicSearchObject publicSearchObject = new PublicSearchObject();
+                  ITransformedNetwork transformedNetwork = transformationManager
+                    .getSearchTransformedNetwork(project.getProjectId(), conceptId, INetworkStatus.APPROVED);
+                  if (transformedNetwork != null && transformedNetwork.getNodes().size() > 0) {
+                      publicSearchObject.setNodes(getNodes(new ArrayList<Node>(transformedNetwork.getNodes().values())));
+                      publicSearchObject.setLinks(getLinks(transformedNetwork.getLinks()));
+                  }
+                  else{
+                      publicSearchObject.setNetworkEmpty(true);
+                  }
+                  return publicSearchObject;
+              };
+             Future<PublicSearchObject> future = taskExecutor.submit(callable);
+             cache.put(new Element(conceptIdToken, future));
+        }
+        return conceptIdToken;
+    }
+    
+    @Override
+    public PublicSearchObject getSearchTransformedNetwork(String conceptIdToken){
+        
+        PublicSearchObject publicSearchObject = new PublicSearchObject();
+        if(!cache.isKeyInCache(conceptIdToken)){
+            publicSearchObject.setStatus(2);
+            return publicSearchObject;
+        }
+        Future<PublicSearchObject> futureResult = (Future<PublicSearchObject>) cache.get(conceptIdToken).getObjectValue();
+        
+        if (futureResult != null && futureResult.isDone()) {
+            try {
+                publicSearchObject = futureResult.get();
+                publicSearchObject.setStatus(1);
+            } catch (InterruptedException | ExecutionException e) {
+                logger.error("Exception while retrieving the result", e);
+                publicSearchObject.setStatus(2);
+            }
+        } else {
+            publicSearchObject.setStatus(3);
```

Comment:
integers are almost always bad status indicator? what does 1, 2, or 3 mean? so hard to maintain.

Reply:
Can I use an Enum while setting the status...to improve the code readability?

Reply:
enums are always a good idea if you have only a limited number of choices :)